reflex/.templates/web/utils/state.js

```diff
@@ -206,7 +203,17 @@ export const applyRestEvent = async (event) => {
  * @param socket The socket object to send the event on.
  */
 export const queueEvents = async (events, socket) => {
-  event_queue.push(...events)
+  event_queue.push(...events.map((ev) => {
+    // Copy the event and set the token.
+    const new_event = {...ev, token: getToken()}
+    // Do NOT overwrite router_data if the event already has it set.
+    if (ev.router_data === undefined || Object.keys(ev.router_data).length === 0) {
+      new_event.router_data = (({ pathname, query, asPath }) => ({ pathname, query, asPath }))(Router)
+    }
+    return new_event
+  }))
```

Comment:
consider removing this section to avoid behavior change;

do not overwrite existing router_data, but continue to set router_data right before the event is sent so that `get_current_page()` is more likely to return the actual page the user is on when the event was sent.